### ISSUE TITLE
Scripts: `plugin-zip` command does not exclude files not included in `files` field

### ISSUE THREAD

**Issue by t-hamano** [WordPress/gutenberg#61171]
### Description

Originally reported in [this comment](https://github.com/WordPress/gutenberg/issues/47660#issuecomment-2079945759).

The `plugin-zip` command refers to the `files` field in `package.json` and allows you to include directories and files other than the default in the zip file:

[@wordpress/scripts – Block Editor Handbook > plugin-zip](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-scripts/#plugin-zip)

However, if this field is defined, it seems that both `package.json` and `readme.txt` are always included in the zip. This means that if you want to include non-default assets in the plugin-zip command, these two files will always be included. Especially when it comes to `package.json`, you often don't want to include it when you want to publish your plugin.

I tried defining package.json as below, but the result did not change.

```json
{
	"name": "test-block",
	"scripts": {
		"plugin-zip": "wp-scripts plugin-zip"
	},
	"devDependencies": {
		"@wordpress/scripts": "^27.7.0"
	},
	"files": []
}

```

```json
{
	"name": "test-block",
	"scripts": {
		"plugin-zip": "wp-scripts plugin-zip"
	},
	"devDependencies": {
		"@wordpress/scripts": "^27.7.0"
	},
	"files": [ "!package.json" ]
}

```

```json
{
	"name": "test-block",
	"scripts": {
		"plugin-zip": "wp-scripts plugin-zip"
	},
	"devDependencies": {
		"@wordpress/scripts": "^27.7.0"
	},
	"files": [ "!readme.txt" ]
}
```

Log output:

```
npm run plugin-zip

> plugin-zip
> wp-scripts plugin-zip

Creating archive for `test-block` plugin... 🎁

Using the `files` field from `package.json` to detect files:

  Adding `package.json`.
  Adding `readme.txt`.

Done. `test-block.zip` is ready! 🎉
```

### Step-by-step reproduction instructions

- Run `npx @wordpress/create-block test-block`
- `cd test-block`
- Update the `files` field in `package.json`
- Run `npm run plugin-zip`

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by danielpost** [comment]
I just ran into the same issue, would be good to be able to ignore files in the root directory.

**Comment by medavidallsop** [comment]
I am also running into this issue, can't figure out a way to exclude `package.json` from the zip.

I discussed this on Slack with @BrianHenryIE and he thinks the root of this issue maybe: https://github.com/npm/npm-packlist/blob/53b2a4f42b7fef0f63e8f26a3ea4692e23a58fed/lib/index.js#L87 (just to provide some extra info that might be useful)

**Comment by arteality** [comment]
same problem

**Comment by t-hamano** [comment]
After further investigation into this issue, I'm unable to say for certain whether this is a bug or not.

The `plugin-zip` command depends on the [npm-packlist](https://github.com/npm/npm-packlist) library. The README states the following:

> If a `package.json` file is found, and it has a `files` list, then ignore everything that isn't in `files`.  **Always include the root readme, license, licence and copying files, if they exist, as well as the package.json file itself.** Non-root readme, license, licence and copying files are included by default, but can be excluded using the  `files` list e.g. `"!readme"`.

In other words, the current behavior seems to be based on the library's specifications.

**Comment by pbking** [comment]
I don't believe there is a fix short of changing how the .zip is built to use something other than wp-scripts.  My understanding is that it isn't a bug either.

I mean, it's not cool, but it's working as the tool was built.  And I don't think refactoring how that is done is the best use of our time.

I'm keen to close this.
